File: PoseKit/Tests/PoseKitTests/Fixtures.swift
import PoseKit

enum Fixtures {
    /// Neutral standing pose, normalized [0,1], y-down. All 19 joints.
    static let standing = PoseVector(points: [
        .nose: [0.5, 0.14], .leftEye: [0.53, 0.12], .rightEye: [0.47, 0.12],
        .leftEar: [0.555, 0.135], .rightEar: [0.445, 0.135],
        .neck: [0.5, 0.22],
        .leftShoulder: [0.585, 0.235], .rightShoulder: [0.415, 0.235],
        .leftElbow: [0.615, 0.36], .rightElbow: [0.385, 0.36],
        .leftWrist: [0.63, 0.475], .rightWrist: [0.37, 0.475],
        .root: [0.5, 0.5], .leftHip: [0.555, 0.5], .rightHip: [0.445, 0.5],
        .leftKnee: [0.55, 0.68], .rightKnee: [0.45, 0.68],
        .leftAnkle: [0.55, 0.86], .rightAnkle: [0.45, 0.86]
    ])

    /// Apply an affine transform (scale, rotation radians, translation) to every point.
    static func transformed(_ pose: PoseVector, scale: Float = 1, rotation: Float = 0,
                            translation: SIMD2<Float> = .zero) -> PoseVector {
        let c = cosf(rotation), s = sinf(rotation)
        var out: [Joint: SIMD2<Float>] = [:]
        for (j, p) in pose.points {
            let r = SIMD2<Float>(c * p.x - s * p.y, s * p.x + c * p.y)
            out[j] = r * scale + translation
        }
        return PoseVector(points: out)
    }
}

#if canImport(Foundation)
import Foundation
#endif

File: android/posekit/src/test/kotlin/com/oerol/posekit/Fixtures.kt
package com.oerol.posekit

import kotlin.math.cos
import kotlin.math.sin

object Fixtures {
    /** Neutral standing pose, normalized [0,1], y-down. All 19 joints. */
    val standing = PoseVector(
        mapOf(
            Joint.nose to Vec2(0.5f, 0.14f),
            Joint.leftEye to Vec2(0.53f, 0.12f),
            Joint.rightEye to Vec2(0.47f, 0.12f),
            Joint.leftEar to Vec2(0.555f, 0.135f),
            Joint.rightEar to Vec2(0.445f, 0.135f),
            Joint.neck to Vec2(0.5f, 0.22f),
            Joint.leftShoulder to Vec2(0.585f, 0.235f),
            Joint.rightShoulder to Vec2(0.415f, 0.235f),
            Joint.leftElbow to Vec2(0.615f, 0.36f),
            Joint.rightElbow to Vec2(0.385f, 0.36f),
            Joint.leftWrist to Vec2(0.63f, 0.475f),
            Joint.rightWrist to Vec2(0.37f, 0.475f),
            Joint.root to Vec2(0.5f, 0.5f),
            Joint.leftHip to Vec2(0.555f, 0.5f),
            Joint.rightHip to Vec2(0.445f, 0.5f),
            Joint.leftKnee to Vec2(0.55f, 0.68f),
            Joint.rightKnee to Vec2(0.45f, 0.68f),
            Joint.leftAnkle to Vec2(0.55f, 0.86f),
            Joint.rightAnkle to Vec2(0.45f, 0.86f),
        )
    )

    /** Apply an affine transform (scale, rotation radians, translation) to every point. */
    fun transformed(
        pose: PoseVector,
        scale: Float = 1f,
        rotation: Float = 0f,
        translation: Vec2 = Vec2.ZERO,
    ): PoseVector {
        val c = cos(rotation)
        val s = sin(rotation)
        val out = mutableMapOf<Joint, Vec2>()
        for ((j, p) in pose.points) {
            val r = Vec2(c * p.x - s * p.y, s * p.x + c * p.y)
            out[j] = r * scale + translation
        }
        return PoseVector(out)
    }
}
